controllers/volumereplication_controller.go

```diff
@@ -126,14 +132,17 @@ func (r *VolumeReplicationReconciler) Reconcile(ctx context.Context, req ctrl.Re
 	nameSpacedName := types.NamespacedName{Name: instance.Spec.DataSource.Name, Namespace: req.Namespace}
 	switch instance.Spec.DataSource.Kind {
 	case pvcDataSource:
-		_, pv, err := r.getPVCDataSource(logger, nameSpacedName)
-		if err != nil {
+		_, pv, pvErr = r.getPVCDataSource(logger, nameSpacedName)
```

Comment:
Fail fast, check if deletion time is zero (IOW, not deleted) and error out immediately.

Possible better readability of code. Code as it exists also works.